connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/ConnectorUtil.java

```diff
@@ -86,4 +88,65 @@ public static InboundConnectorConfiguration getRequiredInboundConnectorConfigura
                         "InboundConnectorExecutable %s is missing @InboundConnector annotation",
                         cls)));
   }
+
+  private static String toNormalizedConnectorName(final String connectorName) {
+    return connectorName.trim().replaceAll("[^a-zA-Z0-9_ ]", "").replaceAll(" ", "_").toUpperCase();
+  }
+
+  private static String toConnectorTypeEnvVariable(final String normalizedConnectorName) {
+    return "CONNECTOR_" + normalizedConnectorName + "_TYPE";
+  }
+
+  private static String toConnectorTimeoutEnvVariable(final String normalizedConnectorName) {
+    return "CONNECTOR_" + normalizedConnectorName + "_TIMEOUT";
+  }
+
+  public static String[] getInputVariables(Class<?> cls, OutboundConnector annotation) {
+    List<ReflectionUtil.MethodWithAnnotation<Operation>> operations =
+        ReflectionUtil.getMethodsAnnotatedWith(cls, Operation.class);
+    if (operations.isEmpty()) {
+      return annotation.inputVariables();
+    } else {
+      return getInputVariables(operations).toArray(new String[0]);
+    }
+  }
+
+  public static Set<String> getInputVariables(
+      List<ReflectionUtil.MethodWithAnnotation<Operation>> operations) {
+    Set<String> variables = new HashSet<String>();
+    operations.forEach(
+        method -> {
+          method
+              .parameters()
+              .forEach(
+                  parameter -> {
+                    if (parameter.getAnnotation(io.camunda.connector.api.annotation.Variable.class)
```

Comment:
Do we also want to handle the case when `@Variable` is not present? e.g. try to use the actual parameter names if they are available (if compiled with the `-parameters` flag).

If not, we can make this part easier to read by inverting the if statement
`if (parameter.getAnnotation(Variable.class) == null) { return; }`